src/routes/dashboard/admin/events/eventsHelper.ts

```diff
@@ -23,28 +24,109 @@ export function convertSaveEventData(data: DashboardEvent): SetAdminEvent {
         discord_url:           trimOrNull(data.discord_url),
         twitch_url:            trimOrNull(data.twitch_url),
         presskit_url:          trimOrNull(data.presskit_url),
-        trailer_youtube_id:    trimOrNull(data.trailer_youtube_id),
+        trailer_url:           trimOrNull(data.trailer_url),
+        trailer_poster_url:    trimOrNull(data.trailer_poster_url),
         description_headline:  data.description_headline.trim(),
         description:           data.description.trim(),
         schedule_visible_from: checkSQLTimeAndDate(convertTimeAndDateToSQL(data.schedule_visible_from)),
         publish_date:          checkSQLTimeAndDate(convertTimeAndDateToSQL(data.publish_date)),
         call_for_papers_start: checkSQLTimeAndDate(convertTimeAndDateToSQL(data.call_for_papers_start)),
         call_for_papers_end:   checkSQLTimeAndDate(convertTimeAndDateToSQL(data.call_for_papers_end)),
+        frontpage_date:        checkSQLTimeAndDate(convertTimeAndDateToSQL(data.frontpage_date)),
     };
 }
 
-export function convertSaveSpeakerData(allSpeaker: DashboardAllEventSpeaker): SetAllAdminEventSpeaker {
+export function convertSaveSpeakerData(allSpeaker: DashboardAllEventSpeaker | undefined): SetAllAdminEventSpeaker {
+    if (!allSpeaker) {
+        return [];
+    }
+
     const result: SetAllAdminEventSpeaker = [];
     for (const speaker of allSpeaker) {
         result.push({
-                        id:           speaker.id,
-                        name:         speaker.name,
-                        visible_from: checkSQLTimeAndDate(convertTimeAndDateToSQL(speaker.visible_from)),
-                    });
+            id:           speaker.id,
+            name:         speaker.name,
+            visible_from: checkSQLTimeAndDate(convertTimeAndDateToSQL(speaker.visible_from)),
+        });
     }
     return result;
 }
 
+export function validateData(data: SetAdminEvent, allSpeaker: SetAllAdminEventSpeaker): string[] {
+    const errorList: string[] = [];
+
+    // text
+    if (data.title.trim().length === 0) {
+        errorList.push('Das Feld "Titel" muss angegeben werden.');
+    }
+    if (data.subtitle.trim().length === 0) {
+        errorList.push('Das Feld "Untertitel" muss angegeben werden.');
+    }
+    if (data.description_headline.trim().length === 0) {
+        errorList.push('Das Feld "Überschrift Beschreibung" muss angegeben werden.');
+    }
+    if (data.description.trim().length === 0) {
+        errorList.push('Das Feld "Beschreibung" muss angegeben werden.');
+    }
+
+    // dates
+    if (!isBeforeOrSameDatesString(data.start_date, data.end_date)) {
+        errorList.push('Das Start-Datum liegt nach dem End-Datum.');
+    }
+
+    if (data.publish_date && data.schedule_visible_from) {
+        if (!isBeforeOrSameDatesString(data.publish_date, data.schedule_visible_from)) {
+            errorList.push('Der Ablaufplan ist vor dem Event sichtbar.');
+        }
+    }
```

Comment:
Can be simplified.
```suggestion
    if (data.publish_date && data.schedule_visible_from
        && !isBeforeOrSameDatesString(data.publish_date, data.schedule_visible_from)
    ) {
        errorList.push('Der Ablaufplan ist vor dem Event sichtbar.');
    }
```
The same applies to the other `if`s, too.

Reply:
I would argue that those nested `if`s are never more readable, because when reading it, the first thought will always be: Why is this nested? Effectively, your only checking a chain of conditions, evaluated in order. The inner `if` (just as a control structure) serves no purpose here. If it's about the readabity, maybe just put every condition in its own line, say:
```ts
    if (
        data.publish_date
        && data.schedule_visible_from
        && !isBeforeOrSameDatesString(data.publish_date, data.schedule_visible_from)
    ) {
        errorList.push('Der Ablaufplan ist vor dem Event sichtbar.');
    }
```
I think this is at least as readable as the original version, but it's not confusion anymore (because nobody will wonder why there are *two* instead of one control structures).

Regarding the other thing you wrote: I don't think you'd need an explicit `null` check. If you're really sure that something is not `null`, you should be able to use the null assertion operator (`!`).